scarlet/resampling.py

```diff
@@ -76,61 +44,91 @@ def match_patches(shape_hr, shape_lr, wcs_hr, wcs_lr):
     assert wcs_hr != None
     assert wcs_lr != None
 
-    im_hr = np.zeros((Ny_hr, Nx_hr))
-    im_lr = np.zeros((Ny_lr, Nx_lr))
 
-    # Coordinates of pixels in both frames
-    y_hr, x_hr = np.where(im_hr == 0)
-    Y_lr, X_lr = np.where(im_lr == 0)
+
+    if psf is True:
+        # Coordinates of all high resolution pixels. This is only needed for PSF matches.
+        y_hr, x_hr = np.indices((Ny_hr, Nx_hr))
+
+        x_hr = x_hr.flatten()
+        y_hr = y_hr.flatten()
+    else:
+        # Coordinates of the high resolution pixels
+        y_hr, x_hr = np.array(range(Ny_hr)), np.array(range(Nx_hr))
+
+    # Capital letters are for coordinates of low-resolution pixels
+    if (isrot is True) or (psf is True):
+
+        # Coordinates of all low resolution pixels. All are needed if frames are rotated.
+        Y_lr, X_lr = np.indices((Ny_lr, Nx_lr))
+
+        X_lr = X_lr.flatten()
+        Y_lr = Y_lr.flatten()
+
+    else:
+        Y_lr, X_lr = np.array(range(Ny_lr)), np.array(range(Nx_lr))
 
     #Corresponding angular positions
+    #of low resolution pixels
     if np.size(wcs_lr.array_shape) == 2:
         ra_lr, dec_lr = wcs_lr.all_pix2world(X_lr, Y_lr, 0, ra_dec_order=True)
     elif np.size(wcs_lr.array_shape) == 3:
         ra_lr, dec_lr = wcs_lr.all_pix2world(X_lr, Y_lr, 0, 0, ra_dec_order=True)
+    #of high resolution pixels
     if np.size(wcs_hr.array_shape) == 2:
-        ra_hr, dec_hr = wcs_hr.all_pix2world(y_hr, x_hr, 0, ra_dec_order=True)
+        ra_hr, dec_hr = wcs_hr.all_pix2world(x_hr, y_hr, 0, ra_dec_order=True)
     elif np.size(wcs_hr.array_shape) == 3:
-        ra_hr, dec_hr = wcs_hr.all_pix2world(y_hr, x_hr, 0, 0, ra_dec_order=True)
+        ra_hr, dec_hr, eggbenedict = wcs_hr.all_pix2world(x_hr, y_hr, 0, 0, ra_dec_order=True)
```

Comment:
But it looks like this is new. You used to only expect two parameters to be returned, even if there were 3 dimensions. Was it just a typo that was untested before (which explains why x and y were reversed)?

Reply:
Indeed that part was untested because the HST images I used in the notebooks don't have a third axis. I made these changes after starting to write tests that made me realise these mistakes.